sim/battle.ts

```diff
@@ -2934,7 +2934,7 @@ export class Battle {
 
 		// Discard remaining secret data
 		// Note: the last \n? is for secret data that are empty when shared
-		return data.replace(/\n\|split\|(?:[^\n]*)\n(?:[^\n]*)\n\n?/g, '\n');
+		return data.replace(/(?<=\n)\|split\|(?:[^\n]*)\n(?:[^\n]*)\n\n?/g, '');
```

Comment:
Do you know why this is the only regexp in this function that's affected? (They are so similar that it seems odd to me that only one would be affected.)
I was also wondering whether using `^` with the `m` flag would be more readable?

Reply:
Only the final regexes are affected because the others don't have the conditional replacement of the last `\n`, which seems to be problematic. I still invite people to mess around with the regexes on their own to verify this a bit more, since regexes tend to be funky.

As for using `m`, did you have something like `^\|split\|(?:[^\n]*)$\n^(?:[^\n]*)$\n(^$\n)?/gm` in mind?